### PR TITLE
refactor: disclosure list item icon placement to right

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksrefactor-mobile-menu-icon-placement-hashicorp.vercel.app/boundary) 🔎
- [Asana task](https://app.asana.com/0/1203652232454021/1204303889362367) 🎟️
- [Figma](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?node-id=815-40857&t=BLzfEIgRtHWvgFn3-0) 🎨 

## 🗒️ What

This PR refactors the placement of the dropdown disclosure item icon to the right hand side. 

Screenshot 2023-04-03 at 5 58 52 PM
Screenshot 2023-04-03 at 5 59 04 PM

## 🧪 Testing

- [Visit the preview](https://dev-portal-git-ksrefactor-mobile-menu-icon-placement-hashicorp.vercel.app/boundary)
- Sign in
- Test the desktop dropdown disclosure placement
- Test the tablet / mobile view

## 💭 Anything else?

We've removed all the icons from the menus except the external link icon. [This is a follow up task.](https://github.com/hashicorp/dev-portal/pull/1793#issuecomment-1491047999)

### CODE REVIEW

File: src/components/dropdown-disclosure/components/list-item/index.tsx [hunk 1]
```diff
@@ -97,10 +97,10 @@ const DropdownDisclosureButtonItem = ({
 					toggleDisclosure()
 				}}
 			>
-				{icon}
 				<Text asElement="span" size={200} weight="medium">
 					{children}
 				</Text>
+				{icon}
```
Comment (kendallstrautman) [inline src/components/dropdown-disclosure/components/list-item/index.tsx:103]: Need to validate whether I can globally change this icon placement. I _think_ its okay.

**Review by zchsh — APPROVED** [review]
✅ Followed the testing steps, seeing everything as described 👍

File: src/components/dropdown-disclosure/components/list-item/index.tsx (hunk 1 above)
Comment (kendallstrautman) [reply src/components/dropdown-disclosure/components/list-item/index.tsx:103]: Ah yes, these two places are where we want the icon to flip so thats good!

**Review by braydenlove — APPROVED** [review]
LGTM

**Review by braydenlove — APPROVED** [review]
Looks good! 🚢 it!

**Comment by kendallstrautman** [comment]
@ashleemboyer I agree I think that could eventually be a nice thing to add. Since we only need it on one side for this work, I'd prefer that to be a follow up task given my current timeframe if possible.

**Comment by kendallstrautman** [comment]
@ashleemboyer i'm going to merge this in but I added a[ follow up ticket](https://app.asana.com/0/1203652232454021/1204332555573341) to capture the icon position idea and added to our eng backlog 😄

**Pull request merged**